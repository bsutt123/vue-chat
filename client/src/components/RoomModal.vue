<template>
  <div :class='$style.container'>
    <button
      class="btn"
      @click="open"
      :class="$style.newRoomButton"
    >
      Create a new Room
    </button>
    <modal
      name="new-room"
      :adaptive="true"
      :width="300"
      :height="200"
    >
      <div :class="$style.modalContainer">
        <h3 :class="$style.modalHeading"> Name your new room </h3>
        <input
          :class="$style.newRoomInput"
          type="text"
          v-model="newRoomName"
        />
        <button
          @click="close"
          :class="$style.modalButton"
          class="btn"
        >
          Close Modal
        </button>
        <button
          @click="addRoom(newRoomName)"
          :class="$style.modalButton"
          class="btn"
        >
          Add Room
        </button>
      </div>
    </modal>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex';

const { mapActions } = createNamespacedHelpers('rooms');
export default {
  name: 'RoomModal',
  data() {
    return {
      newRoomName: '',
    };
  },
  methods: {
    ...mapActions([
      'addRoom',
    ]),
    open() {
      this.$modal.show('new-room');
    },
    close() {
      this.$modal.hide('new-room');
    },
  },
};
</script>

<style lang="postcss" module>

.modalContainer {
  text-align: center;
  padding: 1rem;
  color: black;
}

.modalHeading {
  margin-bottom: 1rem;
}

.newRoomInput {
  width: 90%;
  display: block;
  color: black;
  margin: * auto .5rem auto;
}

.modalButton {
  display: inline-block;
  padding: .25rem .5rem;
  font-size: .8rem;
  border-radius: 2px;
  box-shadow: none;
}

</style>
