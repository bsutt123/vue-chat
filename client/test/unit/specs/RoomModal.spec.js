import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VModal from 'vue-js-modal';

import { CssModuleTestHelperMixin, show, hide } from '../utils/test-helper';
import storeObject from '../utils/testStore';
import RoomModal from '@/components/RoomModal';


const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VModal);
localVue.use(CssModuleTestHelperMixin);

describe('MessageList', () => {
  let store;
  let wrapper;

  beforeEach( () => {
    store = new Vuex.Store(storeObject);
    wrapper = shallow(RoomModal, { localVue, store })
  });

  it('renders the room modal', () => {
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('opens executes after the click event', () => {
    const open = jest.fn()
    wrapper.setMethods({ open })
    wrapper.find('.newRoomButton').trigger('click');
    expect(open.mock.calls.length).toEqual(1);
  });

  it('closes the modal', () => {
    wrapper.find('.newRoomButton').trigger('click')
  })
});