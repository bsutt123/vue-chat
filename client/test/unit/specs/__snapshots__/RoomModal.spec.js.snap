// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList renders the room modal 1`] = `
<div class="container">
    <button class="btn newRoomButton">
        Create a new Room
    </button>
    <!---->
</div>
`;
